homeassistant/components/azure_event_hub/config_flow.py

```diff
@@ -0,0 +1,200 @@
+
+    VERSION: int = 1
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self._data: dict[str, Any] = {}
+        self._options: dict[str, Any] = DEFAULT_OPTIONS
+        self._conn_string: bool | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return AEHOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial user step."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        if user_input is None:
+            return self.async_show_form(step_id=STEP_USER, data_schema=BASE_SCHEMA)
+
+        self._conn_string = user_input.pop(CONF_USE_CONN_STRING)
+        self._data = user_input
+
+        if self._conn_string:
+            return await self.async_step_conn_string()
+        return await self.async_step_sas()
+
+    async def async_step_conn_string(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the connection string steps."""
+        errors = await self.async_update_and_validate_data(user_input)
+        if user_input is None or errors is not None:
+            return self.async_show_form(
+                step_id=STEP_CONN_STRING,
+                data_schema=CONN_STRING_SCHEMA,
+                errors=errors,
+                description_placeholders=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+                last_step=True,
+            )
+
+        return self.async_create_entry(
+            title=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+            data=self._data,
+            options=self._options,
+        )
+
+    async def async_step_sas(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the sas steps."""
+        errors = await self.async_update_and_validate_data(user_input)
+        if user_input is None or errors is not None:
+            return self.async_show_form(
+                step_id=STEP_SAS,
+                data_schema=SAS_SCHEMA,
+                errors=errors,
+                description_placeholders=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+                last_step=True,
+            )
+
+        return self.async_create_entry(
+            title=self._data[CONF_EVENT_HUB_INSTANCE_NAME],
+            data=self._data,
+            options=self._options,
+        )
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import config from configuration.yaml."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        if CONF_SEND_INTERVAL in import_config:
+            self._options[CONF_SEND_INTERVAL] = import_config.pop(CONF_SEND_INTERVAL)
```

Comment:
We're modifying the default options constant here. We shouldn't modify a constant.

Reply:
good one, added deepcopy in init.